WordPress/src/main/java/org/wordpress/android/ui/publicize/adapters/PublicizeServiceAdapter.java

```diff
@@ -118,9 +117,13 @@ public void onBindViewHolder(final SharingViewHolder holder, int position) {
                 mConnections.getServiceConnectionsForUser(mCurrentUserId, service.getId());
 
         holder.mTxtService.setText(service.getLabel());
-        String iconUrl = PhotonUtils.getPhotonImageUrl(service.getIconUrl(), mBlavatarSz, mBlavatarSz);
-        mImageManager.load(holder.mImgIcon, ImageType.BLAVATAR, iconUrl);
-
+        final PublicizeServiceIcon icon = PublicizeServiceIcon.fromServiceId(service.getId());
+        if (icon != null) {
+            holder.mImgIcon.setVisibility(View.VISIBLE);
+            mImageManager.load(holder.mImgIcon, icon.getIconResId());
+        } else {
+            holder.mImgIcon.setVisibility(View.INVISIBLE);
```

Comment:
I think this makes sense, but I was also wondering if we should have a generic `PublicizeServiceIcon` fallback item that uses a generic icon, so we don't need to return `null` from `fromServiceId` and instead return the fallback so we can just show that icon when it's an unknown service. I know iOS has something like that.

No need to implement it right now, but just an idea for the future. We can ask the design team for that icon if we think it makes sense to implement this in the future.

Reply:
Good idea 👍  I'd rather have a default placeholder instead of making the view invisible.